cmd/upload.go

```diff
@@ -159,16 +165,25 @@ func NewUploadCommand(options *UploadOptions, metadataProviders []metadata.Provi
 				}
 				return fmt.Errorf("The --branch or -b flag can only be provided when uploading appmaps from within a Git repository")
 			}
+			
+			if mapsetId == nil {
+				mapSet := appland.BuildMapSet(application, scenarioUUIDs).
```

Comment:
Prior to the server changes, this had to be done after the scenarios were uploaded, of course, but it's always seemed a bit backwards to me. 

Does it make any sense to create the mapset first, and add the scenarios to it?